streams/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java

```diff
@@ -119,7 +119,6 @@ public void shouldAddStreamThread() throws Exception {
                     .sorted().toArray(),
                 equalTo(new String[] {"1", "2", "3"})
             );
-            waitForApplicationState(Collections.singletonList(kafkaStreams), KafkaStreams.State.REBALANCING, DEFAULT_DURATION);
```

Comment:
```REBALANCING``` is an intermediate state so this check can fail if the machine running this test case is too fast.

Reply:
@showuon I can revert this change for your PR #9888 if you brings a better solution :)

Reply:
Agree! The test change I have investigated and did much refactor on that. But the synchronization issue is not what I covered. Nice catch!